src/hooks/api/users/useUpdateUser.ts

```diff
@@ -1,18 +1,34 @@
 import { useCallback } from 'react';
 import { UpdateUserRequest, UpdateUserResponse } from './types';
 import usePost from '../_rest/usePost';
+import { mutate } from 'swr';
+import { User } from 'types/User';
 
 export default function useUpdateUser() {
   const updateUser = usePost();
 
   return useCallback(
-    async (username: string, bio: string) => {
+    async (userId: string, username: string, bio: string) => {
       const result = await updateUser<UpdateUserResponse, UpdateUserRequest>(
         '/users/update/info',
         'update user',
         { username, bio }
       );
 
+      // optimistically update both caches
+      mutate(
+        // TODO: use const generator to make reliable mutate key
+        [`/users/get?username=${username}`, 'fetch user'],
+        (user: User) => ({ ...user, username, bio }),
+        false
+      );
+
+      mutate(
+        [`/users/get?user_id=${userId}`, 'fetch user'],
+        (user: User) => ({ ...user, username, bio }),
+        false
+      );
```

Comment:
mutations are how we allow to client to update immediately: https://swr.vercel.app/docs/mutation